### DISCUSSION TITLE
What are the virtualterminal protocol that can run gum smoothly?

### DISCUSSION THREAD

**Discussion by NeveIsa** [charmbracelet/gum#734] [General]
I am trying to implement a virtual terminal for a OLED display connected to a micro controller and I would like to know what are the protocol that can run gum smoothly. I can write stuff to the OLED using a framebuffer and I want to eventually be able to display gum menus on the oled. I am not even sure if what I am asking is the right question here but please help me with pointers to how I could even get started.

Thank you very much.

**Comment by meowgorithm** [comment]
Hi! G is designed to run in a terminal emulator, so I believe you'd need to run or implement a terminal emulator on the microcontroller and then run Gum on that. Also note that you may need to compile gum with something like [Tiny Go](https://tinygo.org) depending on your architecture; this isn't something we officially support, but would love to know if you hit any issues with it.

Does that help?
